orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/load_table.py

```diff
@@ -123,12 +130,47 @@ def _ingest_table(
 
 
 @solid(
-    required_resource_keys={"bigquery_service", "target_hca_dataset", "scratch_config", "data_repo_client"}
+    required_resource_keys={"bigquery_service", "target_hca_dataset", "scratch_config", "data_repo_client", "gcs"},
+    output_defs=[
+        OutputDefinition(name="no_data", is_required=False),
+        OutputDefinition(name="has_data", is_required=False),
+    ]
+)
+def check_has_data(
+    context: AbstractComputeExecutionContext,
+    metadata_fanout_result: MetadataTypeFanoutResult
+) -> Iterable[Output]:
+    scratch_config = context.resources.scratch_config
+    metadata_type = metadata_fanout_result.metadata_type
+    metadata_path = metadata_fanout_result.path
+
+    source_path = f"{scratch_config.scratch_prefix_name}/{metadata_path}/{metadata_type}/"
+    context.log.info(f"Checking for data to load at path {source_path}")
+    blobs = [blob for blob in
+             context.resources.gcs.list_blobs(scratch_config.scratch_bucket_name, prefix=source_path)]
+    non_empty_metadata_file = any([blob.size > 0 for blob in blobs])
+
+    if non_empty_metadata_file > 0:
+        context.log.info(f"{metadata_type} has data to load")
+        yield Output(True, "has_data")
```

Comment:
This is how we conditionally branch in Dagster; one of these outputs will never materialize and any downstream solids that depend on it will never run.